server-spi/src/main/java/org/keycloak/models/credential/dto/PasswordSecretData.java

```diff
@@ -26,9 +40,15 @@ public PasswordSecretData(@JsonProperty("value") String value, @JsonProperty("sa
         }
     }
 
+    /**
+     * Default creator (Secret consists only of a value and a single salt)
+     * @param value hash value
+     * @param salt salt
+     */
     public PasswordSecretData(String value, byte[] salt) {
         this.value = value;
         this.salt = salt;
+        this.additionalParameters = new MultivaluedHashMap<>();
```

Comment:
Could you lazily create the instance on the getter method and avoid creating this instance every time an instance is created? We are working on reducing footprint and small changes like will help.

Reply:
Doing so would require me to give up the final modified on the fields (lazy creation), or alternatively allow the field to be null. Since all the explicitly-modeled fields are final in these types, I would prefer to not break this pattern (principle of least surprise). 

Again, I will defer to your judgement here, but I at least wanted to raise the point.

Reply:
I prefer the lazy creation or maybe you can just initialize as an empty collection when the instance is created using this constructor?